File: src/main/java/uss/code/course/dto/response/CoursesResponse.java
package uss.code.course.dto.response;

import lombok.Builder;

import java.util.List;

import static lombok.AccessLevel.PRIVATE;

@Builder(access = PRIVATE)
public record CoursesResponse(
        List<CourseResponse> courseResponses
) {
    public static CoursesResponse of(final List<CourseResponse> courseResponses) {
        return CoursesResponse.builder()
                .courseResponses(courseResponses)
                .build();
    }
}

File: src/main/kotlin/uss/code/course/dto/response/CoursesResponse.kt
package uss.code.course.dto.response

data class CoursesResponse(
    val courseResponses: List<CourseResponse>,
) {
    companion object {
        fun of(courseResponses: List<CourseResponse>): CoursesResponse {
            return CoursesResponse(courseResponses = courseResponses)
        }
    }
}
